src/components/nice-anim/nice-anim.tsx

```diff
@@ -33,12 +33,25 @@ export class NiceAnim {
    */
   @Prop() triggerDistance: string = '33%';
 
+  /**
+   * Use this class, in case of no support for IntersectionObserver (server side rendering)
+   */
+  @Prop() fallbackCSSClass: string = 'nice-anim';
+
   io: IntersectionObserver;
 
+  hasIOSupport: boolean;
+
+  constructor() {
+    this.hasIOSupport = typeof IntersectionObserver !== 'undefined';
+  }
+
   componentDidLoad() {
-    this.addIntersectionObserver();
-    const animationDistance = this.direction === 'right' || this.direction === 'down' ? '-' + this.animationDistance : this.animationDistance;
-    (this.el.querySelector('.nice-anim') as HTMLElement).style.setProperty('--distance', animationDistance);
+    if (this.hasIOSupport) {
+      this.addIntersectionObserver();
+      const animationDistance = this.direction === 'right' || this.direction === 'down' ? '-' + this.animationDistance : this.animationDistance;
+      (this.el.querySelector('.nice-anim') as HTMLElement).style.setProperty('--distance', animationDistance);
+    }
```

Comment:
```suggestion
    } else {
      this.el.querySelector('.nice-anim').classList.add(`slide-${this.direction}`);
    }
```

Instead of adding a new property, it may be better to just apply the "slide-{direction}" class if this.hasIOSupport is false. That way everything will still animate in the way it was originally intended, just not on intersection.

This could maybe even be extended to just a helper method instead of having it in two places in the code.

Reply:
the thing is.. there are 2 different use cases...
one: u rehydrate a ssr app.. this way, the element should be opacity 0 after the ssr, so that there is no flickering
two: ssr only... the element should be visible straight away, or like with your improved version, slide in instantly.. whats your take on this?

Reply:
hey since it was late for me already I was a bit quick .. so first of all.. thank you for your great component and for sharing it!!

Reply:
The question is, can we tell if it's a SSR only or a rehydration...? If we could, then handling all three situations will work great. Otherwise, we may have some issues. I'm not sure if there's a solid way to check the difference, especially if people use this component without StencilJS.

Also, requiring a property every time may be a pain for people doing SSR. Possibly a global setting could work?

Reply:
I don't know of a way to tell this.. 
when ssring with stencil there is a prop, but I intentionally did not use this so it works with all wc-setups.. 
I think the standard use-case is rehydration... its also what stencil does.. thats why I left the `.nice-anim` as default class.. pure ssr only setups would have to set a `fallbackCSSClass=""` on all the `<nice-anim>` instances... I think thats ok, since its not the main use case...

Reply:
but of course, its your package and there-fore your decision =)

Reply:
That makes sense. The issue with that, is that if the site is pure SSR, the elements will stay hidden by default and will not animate in using the slide-x class. A way around this would be a global setting we could reference in the code called maybe "pureSSR". Then we would have an extra if on the end that checks for that property and adds the slide-x class manually.

I see that fallbackCSSClass does something similar, but at the same time, it breaks the flow of the original component for SSR.

Reply:
yeah! a global setting for all instances would be the best.. whats the best way to achieve this?

Reply:
I thought of two ways:

1.  Have the user define `window.niceAnimSSR = true;` before importing the component.
2. Have another prop called `pureSsr` (pure-ssr) that the user can set to true on the first `nice-anim` they use. That prop will set the `window.niceAnimSSR` for the user, and then the rest of the props will follow the first.

Not sure which is better. Both require documentation in the README.

Reply:
On second thought 1. might be better because it will allow for additional configurations to be added easily without clogging up the original component. Then later, another component could be added called `nice-anim-config`, which would configure all those global variables.

Reply:
Any update on this @flobacher?

Reply:
@Nicell sry.. I am currently in travel/vacation mode, so I will not be very responsive until the beginning of next week. One global config makes sense to me. global vars on the window are not the safest thing to do because of potential nameing-conflicts... so maybe something like this would make sense:
https://k94n.com/es6-modules-single-instance-pattern

Reply:
Sorry to bug you on vacation. That sounds like a good idea. No rush, just checking in! Thanks again.

Reply:
@Nicell I just updated the PR via a config singleton.. please review again.. thank you!

Reply:
@Nicell any update on this?

Reply:
@flobacher I'm looking at this soon. I'll need to test out the configuration to get a feel for how it works. Would be super helpful if you could explain the way you implemented the configuration and why? It looks nice, I'd just like to hear the reasoning behind it. I still have lots to learn :)

Reply:
^ An example code piece of usage would be great.

Reply:
will try to provide this asap.. I already used/tested it in a project where I use nice-anim  =)

Reply:
Were you able to get that example? I do understand what this does now. Does the class persist between each NiceAnim and updates to it in plain TS? I'm unable to test this right now, but probably will tonight. Also I'm wondering if `Build.isBrowser` will error on something that isn't Stencil.

Reply:
@flobacher I'd like to get this merged. So I'm still unsure on if the configuration persists between all NiceAnim instances, and if having `Build.isBrowser` will error on a non-Stencil site. Once I have those figured out I'll be merging this finally 😄